src/dawn/IIR/StencilInstantiation.cpp

```diff
@@ -98,6 +98,18 @@ bool StencilInstantiation::isIDAccessedMultipleStencils(int accessID) const {
   return false;
 }
 
+bool StencilInstantiation::isIDAccessedMultipleMSs(int accessID) const {
+
+  int count = 0;
+  for(const auto& ms : iterateIIROver<MultiStage>(*IIR_)) {
+    if(ms->hasField(accessID)) {
+      if(++count > 1)
```

Comment:
this  is a weird function. Isn't this the same as just
```
if(ms->hasField(accessID)) return true;
```

Reply:
Actually not. It is checking for count > 1, i.e. count >= 2.
I rewrote a bit, hope it's more readable now.